### PR TITLE
Include production/injection update as timestepping event

### PR DESCRIPTION
**Pull request by totto82**
With this change
```
--time-step-after-event-in-days=
```
will also restrict the timestep after `WCONPROD`/`WCONINJE` etc. keywords.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
The cases that fails uses 
```
TUNING
 1  5  7*  1 /
 /
 /
```
and --enable-tuning=true

i.e. with this PR it takes 1 day time steps after WCONINJE/WCONPROD in the SCHEDULE as expected. 

With this change it now takes 1 day time step at the beginnig o

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
Any comments on this? I would really like this to be merged.

**Comment by GitPaean** [comment]
Sorry for the delay. I will have a look at it now.

**Comment by bska** [comment]
We should discuss this with @vkip whose PR #4660 is in the same area and also affects `TSINIT`.

**Comment by GitPaean** [comment]
FYI, 

the jenkins failures do not look problematic. Only due to the change of the time stepping.

**Comment by GitPaean** [comment]
> We should discuss this with @vkip whose PR https://github.com/OPM/opm-simulators/pull/4660 is in the same area and also affects TSINIT.

Yeah. It is relevant  while at the same time, they are targeting different parts. This PR targets at when we should use the tuning information and https://github.com/OPM/opm-simulators/pull/4660  is targeting how to set up the tuning information (defaulted `TSINIT` not counts anymore).

**Comment by GitPaean** [comment]
@totto82 can you explain a little bit what you are trying to fix with this PR? Thanks.

**Comment by GitPaean** [comment]
benchmark please

**Comment by totto82** [comment]
> @totto82 can you explain a little bit what you are trying to fix with this PR?

The way "--time-step-after-event-in-days=" is supposed to work is that it should make it possible to explicitly set the timestep after an event. For some reason WCONPROD/WCONINJE etc was not listed in the event list. 

>  benchmark please

benchmarking this shouldn't make any difference I think since you will need either need to "enable-tuning" and set the EVENT timesteps in the TUNING keyword or use  "time-step-after-event-in-days"

**Comment by totto82** [comment]
> I am fine with the content in this PR and we need to get @bska agreed from his perspective.

@bska Do you have any perspective on this? I would like to get this merged.

**Comment by bska** [comment]
> > I am fine with the content in this PR and we need to get @bska agreed from his perspective.
> 
> @bska Do you have any perspective on this?

I agree that this is okay.  I will however hold off on updating the reference solutions until we've merged OPM/opm-common#3650 and then verified that OPM/opm-common#3614 does not end up making unexpected changes to the simulation results.  Once those two barriers have been passed I will merge this PR, hopefully in the afternoon CEST today (Wednesday).

**Comment by bska** [comment]
> I will however hold off on updating the reference solutions until we've merged [OPM/opm-common#3650](https://github.com/OPM/opm-common/pull/3650) and then verified that [OPM/opm-common#3614](https://github.com/OPM/opm-common/pull/3614) does not end up making unexpected changes to the simulation results. Once those two barriers have been passed I will merge this PR, hopefully in the afternoon CEST today (Wednesday).

We did not encounter any unexpected problems whilst running that process.  I will prepare an update to the reference solutions for this PR now.

**Comment by bska** [comment]
jenkins build this update_data please

**Comment by bska** [comment]
jenkins build this opm-tests=1026 please

**Review by bska — APPROVED** [review]
The new reference solutions have been installed on the CI system.  I'll merge this now to activate support for taking `TUNING(1,10)` into account after `WCONPROD` and `WCONINJE`.

**Comment by blattms** [comment]
mmh, benchmark results were never posted here, too. I did notify Michael about this. Here they are:
```

| Test    | Configuration                                     | Relative |
| ------- | ------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: drogon - Threads: 1                | 0.999    |
| opm-git | OPM Benchmark: drogon - Threads: 8                | 0.997    |
| opm-git | OPM Benchmark: punqs3 - Threads: 1                | 1.019    |
| opm-git | OPM Benchmark: punqs3 - Threads: 8                | 1.006    |
| opm-git | OPM Benchmark: smeaheia - Threads: 1              | 0.998    |
| opm-git | OPM Benchmark: smeaheia - Threads: 8              | 1.002    |
| opm-git | OPM Benchmark: spe10_model_1 - Threads: 1         | 0.991    |
| opm-git | OPM Benchmark: spe10_model_1 - Threads: 8         | 0.997    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1        | 0.996    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8        | 0.995    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1        | 1.007    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8        | 0.984    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 | 0.982    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 | 0.993    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *
```

**Pull request merged**
